### PR TITLE
limit EC consumption to a minimum threshold

### PR DESCRIPTION
**Pull request by blackliner**
New field minResToLeave to prevent totalt EC consumption (loss of vessel control etc.)

### CODE REVIEW

**Comment by ChrisAdderley** [comment]
I'm deferring this to next version because I really need to get these builds out.

**Comment by blackliner** [comment]
No worries, you need any help on that?

File: Source/ModuleSimpleBoiloff.cs [hunk 1]
```diff
@@ -137,7 +141,9 @@ public void DoCatchup()
         {
           if (part.vessel.missionTime > 0.0)
           {
-              if (part.RequestResource("ElectricCharge", coolingCost * TimeWarp.fixedDeltaTime) < coolingCost * TimeWarp.fixedDeltaTime)
+              vessel.GetConnectedResourceTotals(PartResourceLibrary.Instance.GetDefinition("ElectricCharge").id, out double currentEC, out double maxEC);
```
Comment (ChrisAdderley) [inline Source/ModuleSimpleBoiloff.cs]: it looks like you're calling this function a lot, it's somewhat expensive. Any chance of caching in a variable?

**Comment by blackliner** [comment]
How do you know that it's an expensive call? In my opinion, a part.requestresource somehow also needs to crawl through all EC containers, right?

Did you profile it, or is it an assumption? I don't want to seem sarcastic, I'd really like to know.

**Comment by ChrisAdderley** [comment]
I've been informed by people more in the know about the state of the code than either of us, let's say.

**Comment by blackliner** [comment]
Just to clarify, you are talking about the vessel.getConnected... , not the EC ID, right?

I'll try to think about something, but it needs to know about the current available EC. Maybe I find another function.

**Comment by ChrisAdderley** [comment]
Eh, it's probably fine. There's worse bottlenecks in the code for me to optimize.

**Pull request merged**